pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -244,11 +247,119 @@ func (f *Fit) EventsToRegister() []framework.ClusterEventWithHint {
+		return framework.Queue, err
+	}
+
+	if modifiedPod == nil {
+		logger.V(4).Info("an other pod was deleted", "pod", klog.KObj(pod), "otherPod", klog.KObj(originalPod))
+		return framework.Queue, nil
+	}
+
+	if !f.enableInPlacePodVerticalScaling {
+		// If InPlacePodVerticalScaling (KEP 1287) is disabled or the other pod is unscheduled, then PodUpdate event should be ignored.
+		return framework.QueueSkip, nil
+	}
+
+	// Modifications may or may not be relevant. We only care about modifications that
+	// change the other pod's resource request and the resource is also requested by the
+	// pod we are trying to schedule.
+	var isResourceScaleDown bool
+	originalMaxResourceReq, modifiedMaxResourceReq := &framework.Resource{}, &framework.Resource{}
+	originalMaxResourceReq.SetMaxResource(resource.PodRequests(originalPod, resource.PodResourcesOptions{InPlacePodVerticalScalingEnabled: f.enableInPlacePodVerticalScaling}))
+	modifiedMaxResourceReq.SetMaxResource(resource.PodRequests(modifiedPod, resource.PodResourcesOptions{InPlacePodVerticalScalingEnabled: f.enableInPlacePodVerticalScaling}))
+	reqs := resource.PodRequests(pod, resource.PodResourcesOptions{InPlacePodVerticalScalingEnabled: f.enableInPlacePodVerticalScaling})
+LOOP:
+	for rName, rValue := range reqs {
+		if rValue.IsZero() {
+			continue
+		}
+		switch rName {
+		case v1.ResourceCPU:
+			if originalMaxResourceReq.MilliCPU > modifiedMaxResourceReq.MilliCPU {
+				isResourceScaleDown = true
+				break LOOP
+			}
+		case v1.ResourceMemory:
+			if originalMaxResourceReq.Memory > modifiedMaxResourceReq.Memory {
+				isResourceScaleDown = true
+				break LOOP
+			}
+		case v1.ResourceEphemeralStorage:
+			if originalMaxResourceReq.EphemeralStorage > modifiedMaxResourceReq.EphemeralStorage {
+				isResourceScaleDown = true
+				break LOOP
+			}
+		default:
+			if schedutil.IsScalarResourceName(rName) && originalMaxResourceReq.ScalarResources[rName] > modifiedMaxResourceReq.ScalarResources[rName] {
+				isResourceScaleDown = true
+				break LOOP
+			}
+		}
+	}
+
+	if !isResourceScaleDown {
+		if loggerV := logger.V(7); loggerV.Enabled() {
+			// Log more information.
+			loggerV.Info("an other pod got modified where the pod doesn't care", "pod", klog.KObj(pod), "otherPod", klog.KObj(modifiedPod), "diff", cmp.Diff(originalPod, modifiedPod))
+		} else {
+			logger.V(6).Info("an other pod got modified where the pod doesn't care", "pod", klog.KObj(pod), "otherPod", klog.KObj(modifiedPod))
+		}
+		return framework.QueueSkip, nil
+	}
+
+	logger.V(4).Info("the max request resources of an other pod got updated", "pod", klog.KObj(pod), "otherPod", klog.KObj(modifiedPod))
+	return framework.Queue, nil
+}
+
+// isSchedulableAfterNodeChange is invoked whenever a node added or changed. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (f *Fit) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
Actually, we can do some filtering without waiting for preCheck to be removed - 

```go
func (f *Fit) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
  // ...
  if isFitFunc(pod, modifiedNode) { return framework.Queue}
  return framework.QueueSkip
}
```

Reply:
but we need change the func again once the preCheck is removed. if the preCheck is expected to remove in the
next release, I think we can wait it. what do you think?

Reply:
> if the preCheck is expected to remove in the next release

Yes, it's expected, _or hoped_ to be removed shortly. But, we're not sure. As [this comment describes](https://github.com/kubernetes/kubernetes/pull/119177#issuecomment-1839865993), depending on the result from scheduler_perf, we may not be able to remove it soon. It's better to implement a simple one for now.

Reply:
@sanposhiho updated. please review again.